src/Renderer/ThreeExtended/FeaturesUtils.js

```diff
@@ -0,0 +1,145 @@
+
+            Point is over segment P1P2 if :
+                * if the distance, , is inferior to epsilon
+                * and if :  -epsilon ≤ scalar ≤ (||P1P2|| +  epsilon)
+
+                            + (P0) _
+                            |      |
+                            |      |
+             <---scalar---->|    distance
+                            |      |
+                            |      v
+             +-------------------------------+
+            (P1)                            (P2)
+        *********************************************************** */
+
+        const x1 = linePoints[i]._values[0];
+        const y1 = linePoints[i]._values[1];
+        const x2 = linePoints[j]._values[0];
+        const y2 = linePoints[j]._values[1];
+
+        const Xp = x0 - x1;
+        const Yp = y0 - y1;
+
+        const x21 = x2 - x1;
+        const y21 = y2 - y1;
+        const norm = Math.sqrt(x21 * x21 + y21 * y21);
+        const scalar = (Xp * x21 + Yp * y21) / norm;
+
+        if (scalar >= -epsilon && scalar <= norm + epsilon) {
+            const distance = Math.abs(y21 * x0 - x21 * y0 + x2 * y1 - y2 * x1) / norm;
+            if (distance <= epsilon) {
+                return true;
+            }
+        }
+    }
+
+    return false;
+}
+
+function getClosestPoint(point, points, epsilon) {
+    const x0 = point._values[0];
+    const y0 = point._values[1];
+    let squaredEpsilon = epsilon * epsilon;
+    let closestPoint;
+    for (var i = 0; i < points.length; ++i) {
+        const x1 = points[i]._values[0];
+        const y1 = points[i]._values[1];
+        const xP = x0 - x1;
+        const yP = y0 - y1;
+        const n = xP * xP + yP * yP;
+        if (n < squaredEpsilon) {
+            closestPoint = points[i];
+            squaredEpsilon = n;
+        }
+    }
+    return closestPoint;
+}
+
+function pointIsInsidePolygon(point, polygonPoints) {
+    // ray-casting algorithm based on
+    // http://www.ecse.rpi.edu/Homepages/wrf/Research/Short_Notes/pnpoly.html
+
+    const x = point._values[0];
+    const y = point._values[1];
+
+    let inside = false;
+    // in first j is last point of polygon
+    // for each segment of the polygon (j is i -1)
+    for (let i = 0, j = polygonPoints.length - 1; i < polygonPoints.length; j = i++) {
+        const xi = polygonPoints[i]._values[0];
+        const yi = polygonPoints[i]._values[1];
+        const xj = polygonPoints[j]._values[0];
+        const yj = polygonPoints[j]._values[1];
+
+        // isIntersect semi-infinite ray horizontally with polygon's edge
+        const isIntersect = ((yi > y) != (yj > y))
+            && (x < (xj - xi) * (y - yi) / (yj - yi) + xi);
+        if (isIntersect) {
+            inside = !inside;
+        }
+    }
+
+    return inside;
+}
+
+function filterFeatureUnderCoordinate(coordinate, type, coordinates, epsilon, properties, result) {
```

Comment:
`properties` and `result` are only useful because this function modifies directly `result`.
I think it would be better to modify the signature to:
`isFeatureUnderCoordinate(coordinate, featureType, featureCoords, epsilon)`
and make it return `true` or `false`.

Then the caller becomes:
```js
for (const id in feature.featureVertices) {
    const polygon = feature.featureVertices[id];
    if (polygon.extent && !polygon.extent.isPointInside(coordinate, epsilon)) {
        continue;
    }
    const coordinates = feature.coordinates.slice(polygon.offset, polygon.offset + polygon.count);
    if (isFeatureUnderCoordinate(coordinate, feature.type, coordinates, epsilon) {
    	result.push({
    		coordinates,
    		type: feature.type,
    		properties: features.features[id].properties,
    	});
    }
}